### DISCUSSION TITLE
Export IPYNB to PDF is done

### DISCUSSION THREAD

**Discussion by usamabajwa86** [AammarTufail/python_ka_chilla2023#29] [General]
you can follow below mentioned.

you have to install MIKTIX from below mentioned link

https://miktex.org/download

instal pandoc 

https://pandoc.org/installing.html

You can convert an IPython Notebook file (with the .ipynb extension) to a PDF file using the following steps in Visual Studio Code:

Install the nbconvert package using pip:
Copy code
pip install nbconvert
!pip install pyppeteer
!pyppeteer-install

[pandas_01.pdf](https://github.com/AammarTufail/python_ka_chilla2023/files/10362560/pandas_01.pdf)

**Comment by waqarahmad45** [comment]
Good struggle Usama bhai

 I am stuck at the last command step. 

According to You
Copy code
jupyter nbconvert --to pdf filename.ipynb

Replace "filename" with the actual name of your .ipynb file. This will generate a PDF file with the same name as the .ipynb file.

i have converted today's practice file successfully

[pandas_01.pdf](https://github.com/AammarTufail/python_ka_chilla2023/files/10362560/pandas_01.pdf)


I am not getting this point. Can you help me please?

**Reply by ZahidUllah1992** [reply]
jupyter : The term 'jupyter' is not recognized as the name of a cmdlet, function, script 
file, or operable program. Check the spelling of the name, or if a path was included, verify 
that the path is correct and try again.
At line:1 char:1
+ jupyter nbconvert --to pdf test_2.ipynb

**Reply by ZahidUllah1992** [reply]
It doesnot work. I am having an error in terminal. Please let me know

**Reply by mabdullahshakir4** [reply]
File "/Users/abdullah/opt/anaconda3/lib/python3.9/site-packages/nbconvert/utils/pandoc.py", line 52, in pandoc
    check_pandoc_version()
  File "/Users/abdullah/opt/anaconda3/lib/python3.9/site-packages/nbconvert/utils/pandoc.py", line 100, in check_pandoc_version
    v = get_pandoc_version()
  File "/Users/abdullah/opt/anaconda3/lib/python3.9/site-packages/nbconvert/utils/pandoc.py", line 77, in get_pandoc_version
    raise PandocMissing()
nbconvert.utils.pandoc.PandocMissing: Pandoc wasn't found.
Please check that pandoc is installed:

**Reply by mabdullahshakir4** [reply]
I'm getting this error although i installed pandadoc

**Comment by Atifidrees2023** [comment]
Dear Usman your efforts will be really appreciated if you could just take screenshot of each step and share with us thank you
jupyter nbconvert --to pdf test_2.ipynb
It doesnot work. I am having an error in terminal. Please let me know

**Reply by usamabajwa86** [reply]
just do below mentioned steps


you have to install MIKTIX from below mentioned link

https://miktex.org/download

instal pandoc 

https://pandoc.org/installing.html

You can convert an IPython Notebook file (with the .ipynb extension) to a PDF file using the following steps in Visual Studio Code:

Install the nbconvert package using pip:
Copy code
pip install nbconvert
pip install pyppeteer

**Reply by ZahidUllah1992** [reply]
https://pandic.org/installing.html or pandoc.org

**Reply by usamabajwa86** [reply]
pandoc

**Reply by ZahidUllah1992** [reply]
> just do below mentioned steps
> 
> you have to install MIKTIX from below mentioned link
> 
> https://miktex.org/download
> 
> instal pandoc
> 
> https://pandic.org/installing.html
> 
> You can convert an IPython Notebook file (with the .ipynb extension) to a PDF file using the following steps in Visual Studio Code:
> 
> Install the nbconvert package using pip: Copy code pip install nbconvert pip install pyppeteer

Dear Usman,
I have followed this way, but did not worked, Now i am again trying and let me share the error. Further, please let me know where to run the pip command in code or terminal, because last time it did not run in terminal and cmd

**Reply by ZahidUllah1992** [reply]
<img width="698" alt="Screenshot 2023-01-13 080044" src="https://user-images.githubusercontent.com/121865483/212248572-0f88a73c-8e1c-481d-8650-20a76480dfdb.png">

**Reply by usamabajwa86** [reply]
run in code window

**Reply by ZahidUllah1992** [reply]
<img width="416" alt="export failed" src="https://user-images.githubusercontent.com/121865483/212249356-7a35e274-9377-4888-a03f-6dffa6e1fddd.png">
I run this on zoom, I think I am missing some path or something

**Reply by ZahidUllah1992** [reply]
Can you please come on zoom to check

**Reply by usamabajwa86** [reply]
currently i am in office,  i will check in night after today class

**Reply by ZahidUllah1992** [reply]
Sure, yes, it will be appreciated .

**Reply by Saibtain** [reply]
i tried many time this is not working . i am querous why it is not working.
can you tell em an other way

**Reply by ZahidUllah1992** [reply]
@usamabajwa86 please let me know to check the error

**Reply by usamabajwa86** [reply]
i am available right now, how we can communicate

**Reply by ZahidUllah1992** [reply]
send me a zoom link or should i send you
i am sending here, ok
https://us05web.zoom.us/j/89745652322?pwd=SkZ2aGxtazlCTUpycGpoRW9hTVhhQT09

**Reply by usamabajwa86** [reply]
run these commands

!pip install pyppeteer
!pyppeteer-install

**Reply by ZahidUllah1992** [reply]
Jazak ALLAH,
Done
Reinstalled Python
Reinstalled MikTex
pip install nbconvert
pip install jupyter
pip install pyppeteer
the problem is resolved

**Reply by Atifidrees2023** [reply]
Anyone available on zoom for guidance about Ipynb to pdf?

**Reply by usamabajwa86** [reply]
i will be after today's class

**Comment by bakhtawar-ishaq** [comment]
aoa, can you tell how to do this step?
"Change the current working directory to the location of your .ipynb file"

**Comment by ZahidUllah1992** [comment]
Ok

Zahid
________________________________

**Comment by ZahidUllah1992** [comment]
Ye

Zahid
________________________________

**Comment by Saibtain** [comment]
@usamabajwa86 dear kia ap is process ki vedio record kr ka upload kr sakta han shukria

**Reply by usamabajwa86** [reply]
ok, i will try

**Comment by Saibtain** [comment]
@usamabajwa86 for screen recording ap **FastStone** softwre ap getinto pc ki website sa download kr lany.**(https://getintopc.com/softwares/screen-recording/faststone-capture-2021-free-download/)

**Reply by ZahidUllah1992** [reply]
Reinstalled Python
Reinstalled pandoc
Reinstalled MikTex
Restarted the PC

Then opened VSCODE and run these commands
pip install nbconvert
pip install jupyter
pip install pyppeteer
